std/encoding/name_pattern.go

```diff
@@ -315,7 +315,7 @@ func (n Name) Equal(rhs Name) bool {
 	if len(n) == 0 || &n[0] == &rhs[0] {
 		return true // cheap
 	}
-	for i := 0; i < len(n); i++ {
+	for i := range n {
```

Comment:
Why not just `for i, c := range n` and `if !c.Equal(rhs[i])`

Reply:
Hmm. That'll work, but I'll argue in this case `n[i].Equal(rhs[i])` is more obvious than `c.Equal(rhs[i])`.

Reply:
Well, personally I think Go's for loop with one variable is very confusing, so I write like this...